### PR TITLE
Version 2.2.4-SNAPSHOT

### PR DESCRIPTION
**Pull request by tryptichon**
* Throw exception on detecting wrong token instead of propagating the error to the data
listener.

### CODE REVIEW

File: java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java [hunk 1]
```diff
@@ -226,7 +227,13 @@ public void onTextFrame(String payload, boolean finalFragment, int rsv) {
                         }
                     } else {
                         exitOnClose = true;
-                        process(dataListener, payload);
+                        doReconnect = false;
+                        // process(dataListener, payload);
+                        Object message = error.get("message");
+                        onError(new HiroException(
```
Comment (mglusiuk) [inline java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java]: this is rather reserved for ws protocol errors and not graph errors
so far graph errors was always propagated as special message over data

Comment (tryptichon) [reply java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java]: So maybe propagate them to the dataListener and logListener? Or shall we just reject this pull request and leave it the way it was?

Comment (mglusiuk) [reply java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java]: pls do not modify error propagation.
other changes might be applied

Comment (tryptichon) [reply java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java]: Ok. I rolled back to the old way but kept the renaming of the flag.

**Review by mglusiuk — APPROVED** [review]

**Pull request merged**